dispatch/api/views.py

```diff
@@ -248,6 +252,39 @@ def get_queryset(self):
             queryset = queryset.filter(title__icontains=q)
         return queryset
 
+class PollViewSet(DispatchModelViewSet):
+    """Viewset for the Poll model views."""
+    model = Poll
+    serializer_class = PollSerializer
+
+    def get_queryset(self):
+        queryset = Poll.objects.all()
+        q = self.request.query_params.get('q', None)
+        if q is not None:
+            queryset = queryset.filter(Q(name__icontains=q) | Q(question__icontains=q) )
+        return queryset
+
+class PollVoteViewSet(mixins.CreateModelMixin, viewsets.GenericViewSet):
+    """Viewset for the PollVote Model"""
+    permission_classes = (AllowAny,)
+
+    model = PollVote
+    serializer_class = PollVoteSerializer
+
+    def create(self, request):
+        if 'vote_id' in request.data.keys():
+            vote_id = request.data['vote_id']
+            try:
+                vote = PollVote.objects.get(id=vote_id)
+                vote.delete()
+            except PollVote.DoesNotExist:
```

Comment:
this should be equivalent without needing the not-exists try-catch (it will just do nothing if the filtered qs is empty):
```
PollVote.objects.filter(id=vote_id).delete()
```